heron/common/src/java/com/twitter/heron/common/basics/WakeableLooper.java

```diff
@@ -157,6 +182,10 @@ private void executeTasksOnWakeup() {
     // We pre-get the size to avoid execute the tasks added during execution
     int s = tasksOnWakeup.size();
     for (int i = 0; i < s; i++) {
+      // in case previous task cleared the list
```

Comment:
This won't happen during the state restore since I just add the locks which will prevent new tasks from being add during the state restore.  And its only during the state restore that we clear the tasks

Reply:
I think we need to answer two questions beforehand:
- where the tasks are added?
- when can tasks be added?